dex/src/main/scala/com/wavesplatform/dex/cli/Actions.scala

```diff
@@ -0,0 +1,574 @@
+          s"""
+             |Passed arguments:
+             |  DEX REST API          : ${args.dexRestApi}
+             |  Waves Node REST API   : ${args.nodeRestApi}
+             |  Expected DEX version  : ${args.version}
+             |  DEX config path       : ${args.configPath}
+             |  Auth Service REST API : ${args.authServiceRestApi.getOrElse("")}
+             |  Account seed          : ${args.accountSeed.getOrElse("")}
+                   """.stripMargin
+        )
+        superConnector <- SuperConnector.create(matcherSettings, args.dexRestApi, args.nodeRestApi, args.authServiceRestApi, apiKey)
+        checkResult <- new Checker(superConnector).checkState(args.version, args.accountSeed, config, matcherSettings)
+        _ <- cli.lift(superConnector.close())
+      } yield checkResult
+    ) match {
+      case Right(diagnosticNotes) => println(s"$diagnosticNotes\nCongratulations! All checks passed!")
+      case Left(error) => println(error); forceStopApplication(MatcherStateCheckingFailedError)
+    }
+  }
+
+  // noinspection ScalaStyle
+  def runComparison(args: Args): Unit =
+    (for {
+      _ <- cli.log(
+        s"""
+           |Passed arguments:
+           |  DEX config path : ${args.configPath}
+           |Running in background
+           |""".stripMargin
+      )
+      tool <- ComparisonTool(args.configPath)
+      _ <- cli.lift(tool.run()) // TODO logger context
+    } yield ()) match {
+      case Right(_) =>
+      case Left(error) => println(error); forceStopApplication(MatcherStateCheckingFailedError)
+    }
+
+  // noinspection ScalaStyle
+  def makeSnapshots(args: Args): Unit = {
+    cli.log(
+      s"""
+         |Passed arguments:
+         |  DEX config path : ${args.configPath}
+         |  DEX REST API    : ${args.dexRestApi}
+         |  Timeout         : ${args.timeout}
+         |Running in background
+         |""".stripMargin
+    )
+
+    implicit val scheduler: SchedulerService = Scheduler.singleThread(
+      name = "time-impl",
+      daemonic = true,
+      executionModel = ExecutionModel.AlwaysAsyncExecution
+    )
+
+    val key = readSecretFromStdIn("Enter X-API-KEY: ")
+
+    val currentOffset = sendRequest(s"${args.dexRestApi}/matcher/debug/currentOffset", key).toInt
+    sendRequest(s"${args.dexRestApi}/matcher/debug/currentOffset", key, "post")
+
+    val validation = Task(sendRequest(args.dexRestApi, "oldestSnapshotOffset", key).toInt <= currentOffset)
+      .delayExecution(1.second)
+      .onErrorRestart(Long.MaxValue)
+      .restartUntil(_ == true)
+      .timeout(args.timeout)
+      .runToFuture
+
+    Try(Await.ready(validation, args.timeout)) match {
+      case Success(_) => println(s"Success!")
+      case Failure(e) =>
+        e match {
+          case _: TimeoutException => println("Snapshots wasn't saved before reaching timeout")
+          case _ => println(s"Other error happened:\n${e.printStackTrace()}")
+        }
+        System.exit(1)
```

Comment:
forceStopApplication(..) ?

Reply:
We don't need it here, because we stop only CLI and don't have to do anything else to stop it